react/components/AddressCompletionForm.tsx

```diff
@@ -42,9 +44,22 @@ const AddressCompletionForm: React.FC<Props> = ({
   const {
     orderForm: { clientProfileData },
   } = useOrderForm()
+  const { autocompletedAddress } = useOrderShipping()
 
   const [buyerIsReceiver, setBuyerIsReceiver] = useState(true)
-  const { address, invalidFields } = useAddressContext()
+  const { address, invalidFields, rules } = useAddressContext()
+
+  const countryRules = (address.country && rules[address.country]) || undefined
+  const { fields, display } = countryRules ?? {}
+  const mandatoryEditableFields = autocompletedAddress
+    ? Object.entries(fields ?? {})
+        .filter(
+          ([fieldName, fieldSchema]) =>
+            fieldSchema.required &&
+            !autocompletedAddress[fieldName as keyof Address]
+        )
+        .map(([fieldName]) => fieldName)
+    : []
```

Comment:
do you think it would be helpful if we extracted the validation logic for the address to a separate context with low-level functions to validate an address? like having a `useAddressValidator` hook that would expose a `validateAddress` function that would return what the `invalidFields` the `useAddressContext` currently returns.

Reply:
Hm... Where would that context live? Maybe we can simply add this validating function to the OrderShipping context.

Reply:
> Where would that context live?

I was thinking in the same place where we pass the rules to `AddressContext`.

> Maybe we can simply add this validating function to the OrderShipping context.

I thought about a new context because we could theoretically use the address context in other apps right?